### PR TITLE
Microverse Project: Todo List App

### PR DESCRIPTION
**Pull request by happiguru**
This pull request implements the following:-

1. Todo List Application built entirely with JavaScript
2. ES6 and Modules
3. Object Oriented Programming Approach
4. Webpack bundler
5. date-fns

Thank You

### CODE REVIEW

**Review by addod19 — CHANGES_REQUESTED** [review]
## Status: Require changes :recycle: :traffic_light: 

Hi @happiguru, great job on this project :+1: Nice job fixing most of the requirements of the project; - well-written PR, - professional README file, - live link works perfectly, implements ES6 syntax :100:  :smile: However, there are a few comments we need to address before proceeding, kindly check the comments below: :fireworks: 

- The Task Class needs a little improvement; we should have the following fields: title, description, dueDate, and priority. We currently have a name and dueDate.  Let us kindly fix this
- Let us implement the edit functionality of the interface, we are missing it
- In the package.json file, we are missing the "dev" script. Let us kindly add it `"dev": "webpack --mode development",` so we can run commands "npm run dev"
- Kindly add `webpack-dev-server` to our package.json file under devDependencies
- Our project directory is missing a .gitignore file, kindly add it and add the node_modules to the file

Kindly submit another request once these are resolved :smile: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "git.ignoreLimitWarning": true
+}
```
Comment (wathigo) [inline .vscode/settings.json:3]: Files not related to the project should not be committed and added to .gitignore (e.g. .DS_Store) ([screenshot from Pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/gitignore.png))

File: src/javascript/home/header.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+function header() {
```
Comment (wathigo) [inline src/javascript/home/header.js]: It uses ES6 syntax ([screenshot from Pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/todo_list/es6_requirement.png)).
Let's make use of the arrow functions

File: src/javascript/modules/tasks.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+export default class Task {
+  constructor(name, dueDate = 'No date') {
+    this.name = name;
+    this.dueDate = dueDate;
+  }
```
Comment (wathigo) [inline src/javascript/modules/tasks.js]: A todo item should contain the following fields
([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/todo_list/todo_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/todo-list)):
* A title
* A description
* A due date
* A priority

I was not prompted to enter most of this field in an attempt to create a new task.

**Review by wathigo — CHANGES_REQUESTED** [review]
## CHANGES REQUIRED :o: 
Great job implementing the suggested changes.
Let's fix a few more issues before we move on.

### REQUIRED
* There should be a default project ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/todo_list/projects_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/todo-list)). See the comments for more issues :point_down: 

**After making the changes, submit another request for review.**

**Happy Coding :computer: 
Simon Wathigo**

File: .vscode/settings.json (hunk 1 above)
Comment (happiguru) [reply .vscode/settings.json:3]: resolved

**Review by NtwaliHeritier — APPROVED** [review]
### **STATUS: APPROVED✔**

Great work making the required changes.
You can merge👏👏👏👏👏

Happy coding💻💻💻💻💻💻

**Pull request merged**